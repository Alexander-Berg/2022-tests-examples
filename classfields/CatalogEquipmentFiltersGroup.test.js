const React = require('react');
const { shallow } = require('enzyme');
const { shallowToJson } = require('enzyme-to-json');

const CatalogEquipmentFiltersGroup = require('./CatalogEquipmentFiltersGroup');

it('должен отрендерить сначала селекты, потом чекбоксы, причем ровно в том порядке, в котором они пришли', () => {
    // NOTE: чекбоксы и селекты могут идти вперемешку
    const category = {
        name: 'название',
        groups: [
            {
                options: [
                    { code: 'code1', name: 'чекбокс1' },
                    { code: 'code2', name: 'чекбокс2' },
                ],
            },
            { name: 'селект1', options: [] },
            {
                options: [
                    { code: 'code3', name: 'чекбокс3' },
                    { code: 'code4', name: 'чекбокс4' },
                ],
            },
            { name: 'селект2', options: [] },
        ],
    };

    const wrapper = shallow(<CatalogEquipmentFiltersGroup category={ category } onChange={ jest.fn() }/>);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
});
