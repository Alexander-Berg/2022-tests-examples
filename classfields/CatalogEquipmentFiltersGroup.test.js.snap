// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`должен отрендерить сначала селекты, потом чекбоксы, причем ровно в том порядке, в котором они пришли 1`] = `
<div
  className="CatalogEquipmentFiltersGroup CatalogEquipmentFiltersGroup_controlSize_m"
>
  <div
    className="CatalogEquipmentFiltersGroup__title"
  >
    название
  </div>
  <div
    className="CatalogEquipmentFiltersGroup__group"
  >
    <div
      className="CatalogEquipmentFiltersGroup__column"
      key="0"
    >
      <div
        className="CatalogEquipmentFiltersGroup__item"
        key="селект1"
      >
        <CatalogEquipmentFilter
          controlSize="m"
          group={
            Object {
              "name": "селект1",
              "options": Array [],
            }
          }
          onChange={[MockFunction]}
          value={Array []}
        />
      </div>
    </div>
    <div
      className="CatalogEquipmentFiltersGroup__column"
      key="1"
    >
      <div
        className="CatalogEquipmentFiltersGroup__item"
        key="селект2"
      >
        <CatalogEquipmentFilter
          controlSize="m"
          group={
            Object {
              "name": "селект2",
              "options": Array [],
            }
          }
          onChange={[MockFunction]}
          value={Array []}
        />
      </div>
    </div>
    <div
      className="CatalogEquipmentFiltersGroup__column"
      key="2"
    />
  </div>
  <div
    className="CatalogEquipmentFiltersGroup__group"
  >
    <div
      className="CatalogEquipmentFiltersGroup__column"
      key="0"
    >
      <div
        className="CatalogEquipmentFiltersGroup__item"
        key="чекбокс1"
      >
        <CatalogEquipmentFilter
          controlSize="m"
          group={
            Object {
              "code": "code1",
              "name": "чекбокс1",
            }
          }
          onChange={[MockFunction]}
          value={Array []}
        />
      </div>
      <div
        className="CatalogEquipmentFiltersGroup__item"
        key="чекбокс2"
      >
        <CatalogEquipmentFilter
          controlSize="m"
          group={
            Object {
              "code": "code2",
              "name": "чекбокс2",
            }
          }
          onChange={[MockFunction]}
          value={Array []}
        />
      </div>
    </div>
    <div
      className="CatalogEquipmentFiltersGroup__column"
      key="1"
    >
      <div
        className="CatalogEquipmentFiltersGroup__item"
        key="чекбокс3"
      >
        <CatalogEquipmentFilter
          controlSize="m"
          group={
            Object {
              "code": "code3",
              "name": "чекбокс3",
            }
          }
          onChange={[MockFunction]}
          value={Array []}
        />
      </div>
    </div>
    <div
      className="CatalogEquipmentFiltersGroup__column"
      key="2"
    >
      <div
        className="CatalogEquipmentFiltersGroup__item"
        key="чекбокс4"
      >
        <CatalogEquipmentFilter
          controlSize="m"
          group={
            Object {
              "code": "code4",
              "name": "чекбокс4",
            }
          }
          onChange={[MockFunction]}
          value={Array []}
        />
      </div>
    </div>
  </div>
</div>
`;
